src/main/webapp/app/exercises/shared/rating/rating.component.ts

```diff
@@ -23,8 +23,6 @@ export class RatingComponent implements OnInit {
             return;
         }
 
-        // delete participation to prevent circular dependency
-        this.result.participation = undefined;
         this.ratingService.getRating(this.result.id!).subscribe((rating) => {
```

Comment:
Same goes for the exclamation mark here as below
```suggestion
        if (this.result.id !== undefined) {
                this.ratingService.getRating(this.result.id).subscribe((rating) => {
```

Reply:
Done in `a92ca57`